services/data_plane_manager/src/main/java/com/futurewei/alcor/dataplane/service/impl/DpmServiceImpl.java

```diff
@@ -188,11 +189,18 @@ private void patchGoalstateForNeighbor(UnicastGoalState unicastGoalState) throws
                                 goalStateBuilder.removeRouterStates(0);
                             }
 
-                            // Add subnet to router_state
+                            String routerId = subnetEntity.getRouterId();
+                            if (routerId == null) {
+                                List<InternalRouterInfo> internalRouterInfos = networkConfig.getInternalRouterInfos();
+                                for (InternalRouterInfo internalRouterInfo : internalRouterInfos) {
+                                    routerId = internalRouterInfo.getRouterConfiguration().getId();
+                                    if (routerId != null) break;
+                                }
+                            }
```

Comment:
Does this new block of code try to reset routerId? If so, please put some comment to explain the context.

Reply:
I have put comments in the code. 
Basically, this function will be called by two test scenario (#4 and #5). 
Test scenario #4 has routerId inside of subnet, but test scenario #5 doesn't, so test scenario #5 needs get routerId from networkCofnig.